pkg/plugins/policies/meshaccesslog/plugin/v1alpha1/access_log_v2.go

```diff
@@ -0,0 +1,147 @@
+	envoyFormat, err := accesslog.ParseFormat(formatString + "\n")
+
+	if err != nil {
+		return nil, errors.Wrapf(err, "invalid access log format string: %s", formatString)
+	}
+
+	variables := accesslog.InterpolationVariables{
+		accesslog.CMD_KUMA_SOURCE_ADDRESS:              net.JoinHostPort(c.Dataplane.GetIP(), "0"), // deprecated variable
+		accesslog.CMD_KUMA_SOURCE_ADDRESS_WITHOUT_PORT: c.Dataplane.GetIP(),                        // replacement variable
+		accesslog.CMD_KUMA_SOURCE_SERVICE:              c.SourceService,
+		accesslog.CMD_KUMA_DESTINATION_SERVICE:         c.DestinationService,
+		accesslog.CMD_KUMA_MESH:                        c.Mesh,
+		accesslog.CMD_KUMA_TRAFFIC_DIRECTION:           string(c.TrafficDirection),
+	}
+
+	envoyFormat, err = envoyFormat.Interpolate(variables)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to interpolate access log format string with Kuma-specific variables: %s", formatString)
+	}
+
+	return envoyFormat, nil
+}
+
+func (c *Configurer) envoyAccessLog(defaultFormat string) (*envoy_accesslog.AccessLog, error) {
+	var format *api.MeshAccessLog_Format
+	if f := c.Backend.GetFile().GetFormat(); f != nil {
+		format = f
+	} else if f := c.Backend.GetTcp().GetFormat(); f != nil {
+		format = f
+	}
+
+	// TODO json
+
+	if len(format.GetJson()) == 0 {
+		formatString := format.GetPlain()
+		if formatString == "" {
+			formatString = defaultFormat
+		}
+		envoyFormat, err := c.handlePlain(formatString)
+		if err != nil {
+			return nil, err
+		}
+
+		if file := c.Backend.GetFile(); file != nil {
+			return fileAccessLog(envoyFormat.String(), file.Path)
+		} else if tcp := c.Backend.GetTcp(); tcp != nil {
+			path := envoy.AccessLogSocketName(c.Dataplane.Meta.GetName(), c.Mesh)
+			return fileAccessLog(fmt.Sprintf("%s;%s", tcp.Address, envoyFormat.String()), path)
+		}
+	}
+
+	panic("impossible backend type")
+}
+
+func fileAccessLog(format string, path string) (*envoy_accesslog.AccessLog, error) {
+	fileAccessLog := &access_loggers_file.FileAccessLog{
+		AccessLogFormat: &access_loggers_file.FileAccessLog_LogFormat{
+			LogFormat: &envoy_core.SubstitutionFormatString{
+				Format: &envoy_core.SubstitutionFormatString_TextFormatSource{
+					TextFormatSource: &envoy_core.DataSource{
+						Specifier: &envoy_core.DataSource_InlineString{
+							InlineString: format,
+						},
+					},
+				},
+			},
+		},
+		Path: path,
+	}
+
+	marshaled, err := util_proto.MarshalAnyDeterministic(fileAccessLog)
+	if err != nil {
+		return nil, errors.Wrapf(err, "could not marshall %T", fileAccessLog)
+	}
+	return &envoy_accesslog.AccessLog{
+		Name: "envoy.access_loggers.file",
+		ConfigType: &envoy_accesslog.AccessLog_TypedConfig{
+			TypedConfig: marshaled,
+		},
+	}, nil
+}
+
+func (c *Configurer) Configure(filterChain *envoy_listener.FilterChain) error {
+	return listeners_v3.UpdateFilterConfig(filterChain, "envoy.filters.network.http_connection_manager", func(filterConfig proto.Message) error {
```

Comment:
Seems like you do for both TCP and HTTP on the http_connection_manager no?

Maybe you should use: 

https://github.com/michaelbeaumont/kuma/blob/e8b1d012f79138f967b41be45b1d629dcd4de357/pkg/xds/envoy/listeners/v3/util.go#L18-L36

And handle `NewUnexpectedFilterConfigTypeError` (probably needs some typing here) to ignore it.

Reply:
Yeah you're right the "switch" is happening at the wrong level